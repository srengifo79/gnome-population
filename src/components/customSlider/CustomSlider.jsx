import { useState } from "react";
import { Slider } from "@mui/material";
import styled from "styled-components";

import { colors } from "../../theme/colors";

const StyledSlider = styled.div`
  .slider-container {
    text-align: center;

    .MuiSlider-root {
      width 92%; 
      color: ${colors.textBlue};
    }
  }
`;

const CustomSlider = ({ name, range, onSlide }) => {
  const [value, setValue] = useState(range);

  const handleSlide = (e, val) => {
    setValue(val);
    onSlide(val);
  };

  return (
    <StyledSlider>
      <span>{name}</span>
      <div className="slider-container">
        <Slider
          value={value}
          onChange={handleSlide}
          valueLabelDisplay="auto"
          min={range[0]}
          max={range[1]}
        />
      </div>
    </StyledSlider>
  );
};

export default CustomSlider;
