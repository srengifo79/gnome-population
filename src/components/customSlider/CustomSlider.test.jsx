import React from "react";
import { screen, render, cleanup, fireEvent } from "@testing-library/react";
import { toMatchDiffSnapshot } from "snapshot-diff";

import CustomSlider from "./CustomSlider";

expect.extend({ toMatchDiffSnapshot });

afterEach(cleanup);

it("Initial snapshop", () => {
  const { asFragment } = render(
    <CustomSlider name="Age" range={[0, 20]} onSlide={() => {}} />
  );
  expect(asFragment()).toMatchSnapshot();
});

it("Move slide snapshot", () => {
  const mockOnSlide = jest.fn();
  const { asFragment } = render(
    <CustomSlider name="Age" range={[0, 20]} onSlide={mockOnSlide} />
  );

  const initalRender = asFragment();

  const endSlider = screen.getByDisplayValue("20");
  fireEvent.change(endSlider, { target: { value: "10" } });

  expect(endSlider.getAttribute("value")).toBe("10");
  expect(mockOnSlide).toBeCalledWith([0, 10]);
  expect(initalRender).toMatchDiffSnapshot(asFragment());
});
