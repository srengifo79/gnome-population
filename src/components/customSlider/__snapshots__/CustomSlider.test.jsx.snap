// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Initial snapshop 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL ekJtLn"
  >
    <span>
      Age
    </span>
    <div
      class="slider-container"
    >
      <span
        class="MuiSlider-colorPrimary MuiSlider-sizeMedium MuiSlider-root css-187mznn-MuiSlider-root"
      >
        <span
          class="MuiSlider-rail css-14pt78w-MuiSlider-rail"
        />
        <span
          class="MuiSlider-track css-1gv0vcd-MuiSlider-track"
          style="left: 0%; width: 100%;"
        />
        <span
          class="MuiSlider-thumbColorPrimary MuiSlider-thumbSizeMedium MuiSlider-thumb css-eg0mwd-MuiSlider-thumb"
          data-index="0"
          style="left: 0%;"
        >
          <input
            aria-orientation="horizontal"
            aria-valuemax="20"
            aria-valuemin="0"
            aria-valuenow="0"
            data-index="0"
            max="20"
            min="0"
            style="border: 0px; height: 100%; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 100%; direction: ltr;"
            type="range"
            value="0"
          />
          <span
            aria-hidden="true"
            class="MuiSlider-valueLabel css-1kz0hui-MuiSlider-valueLabel"
          >
            <span
              class="MuiSlider-valueLabelCircle"
            >
              <span
                class="MuiSlider-valueLabelLabel"
              >
                0
              </span>
            </span>
          </span>
        </span>
        <span
          class="MuiSlider-thumbColorPrimary MuiSlider-thumbSizeMedium MuiSlider-thumb css-eg0mwd-MuiSlider-thumb"
          data-index="1"
          style="left: 100%;"
        >
          <input
            aria-orientation="horizontal"
            aria-valuemax="20"
            aria-valuemin="0"
            aria-valuenow="20"
            data-index="1"
            max="20"
            min="0"
            style="border: 0px; height: 100%; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 100%; direction: ltr;"
            type="range"
            value="20"
          />
          <span
            aria-hidden="true"
            class="MuiSlider-valueLabel css-1kz0hui-MuiSlider-valueLabel"
          >
            <span
              class="MuiSlider-valueLabelCircle"
            >
              <span
                class="MuiSlider-valueLabelLabel"
              >
                20
              </span>
            </span>
          </span>
        </span>
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Move slide snapshot 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -14,11 +14,11 @@
          <span
            class=\\"MuiSlider-rail css-14pt78w-MuiSlider-rail\\"
          />
          <span
            class=\\"MuiSlider-track css-1gv0vcd-MuiSlider-track\\"
-           style=\\"left: 0%; width: 100%;\\"
+           style=\\"left: 0%; width: 50%;\\"
          />
          <span
            class=\\"MuiSlider-thumbColorPrimary MuiSlider-thumbSizeMedium MuiSlider-thumb css-eg0mwd-MuiSlider-thumb\\"
            data-index=\\"0\\"
            style=\\"left: 0%;\\"
@@ -49,37 +49,37 @@
                </span>
              </span>
            </span>
          </span>
          <span
-           class=\\"MuiSlider-thumbColorPrimary MuiSlider-thumbSizeMedium MuiSlider-thumb css-eg0mwd-MuiSlider-thumb\\"
+           class=\\"MuiSlider-thumbColorPrimary MuiSlider-thumbSizeMedium MuiSlider-thumb Mui-focusVisible css-eg0mwd-MuiSlider-thumb\\"
            data-index=\\"1\\"
-           style=\\"left: 100%;\\"
+           style=\\"left: 50%;\\"
          >
            <input
              aria-orientation=\\"horizontal\\"
              aria-valuemax=\\"20\\"
              aria-valuemin=\\"0\\"
-             aria-valuenow=\\"20\\"
+             aria-valuenow=\\"10\\"
              data-index=\\"1\\"
              max=\\"20\\"
              min=\\"0\\"
              style=\\"border: 0px; height: 100%; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 100%; direction: ltr;\\"
              type=\\"range\\"
-             value=\\"20\\"
+             value=\\"10\\"
            />
            <span
              aria-hidden=\\"true\\"
-             class=\\"MuiSlider-valueLabel css-1kz0hui-MuiSlider-valueLabel\\"
+             class=\\"MuiSlider-valueLabelOpen MuiSlider-valueLabel css-1kz0hui-MuiSlider-valueLabel\\"
            >
              <span
                class=\\"MuiSlider-valueLabelCircle\\"
              >
                <span
                  class=\\"MuiSlider-valueLabelLabel\\"
                >
-                 20
+                 10
                </span>
              </span>
            </span>
          </span>
        </span>"
`;
